File: Solutions/ISHNN/Game Shop/4/Weapon.cpp
#include <iomanip>
#include "Weapon.h"

Weapon::Weapon(std::string name, std::string descripition, int wight, int value, int damage):Item(name,descripition,wight,value)
{
	Weapon::damage = damage;
}


Weapon::~Weapon()
{
}

void Weapon::Describe() {
	Item::Describe();
	std::cout << std::left << std::setw(13) << "Damage" << " = " << damage << std::endl;
}

File: Solutions/DSMdongly/game-shop/part-3/src/game-shop/item/weapon.go
package item

import "fmt"

type Weapon struct {
	Item
	Damage int
}

func NewWeapon(nme string, dsc string, wgt int, val int, dmg int) *Weapon {
	return &Weapon{
		Item:   *NewItem(nme, dsc, wgt, val),
		Damage: dmg,
	}
}

func (wep Weapon) Describe() {
	wep.Item.Describe()
	fmt.Printf("Damage      = %d\n", wep.Damage)
}
